src/RobotInterface/YarpImplementation/src/YarpSensorBridge.cpp

```diff
@@ -52,6 +52,76 @@ bool YarpSensorBridge::initialize(std::weak_ptr<const IParametersHandler> handle
+                                   m_pimpl->metaData.bridgeOptions.isPIDsEnabled);
+    if (!ret)
+    {
+        log()->info("{} Skipping the configuration of configureRemoteControlBoardRemapper. "
+                    "YarpSensorBridge "
+                    "will not stream relevant measures.",
+                    logPrefix);
+    }
+
+    ret = m_pimpl->subConfigLoader("stream_motor_states",
+                                   "Motors",
+                                   &YarpSensorBridge::Impl::configureRemoteControlBoardRemapper,
+                                   handler,
+                                   m_pimpl->metaData,
+                                   m_pimpl->metaData.bridgeOptions.isMotorEncodersEnabled);
+    if (!ret)
+    {
+        log()->info("{} Skipping the configuration of configureRemoteControlBoardRemapper. "
+                    "YarpSensorBridge "
+                    "will not stream relevant measures.",
+                    logPrefix);
+    }
+
+    ret = m_pimpl->subConfigLoader("stream_motor_PWM",
+                                   "MotorPWM",
+                                   &YarpSensorBridge::Impl::configureRemoteControlBoardRemapper,
+                                   handler,
+                                   m_pimpl->metaData,
+                                   m_pimpl->metaData.bridgeOptions.isPWMControlEnabled);
+    if (!ret)
+    {
+        log()->info("{} Skipping the configuration of configureRemoteControlBoardRemapper. "
+                    "YarpSensorBridge "
+                    "will not stream relevant measures.",
+                    logPrefix);
+    }
+
+    ret = m_pimpl->subConfigLoader("stream_joint_torques",
+                                   "WBDEstimates",
+                                   &YarpSensorBridge::Impl::configureRemoteControlBoardRemapper,
+                                   handler,
+                                   m_pimpl->metaData,
+                                   m_pimpl->metaData.bridgeOptions.isWBDEstimatesEnabled);
+    if (!ret)
+    {
+        log()->info("{} Skipping the configuration of configureRemoteControlBoardRemapper. "
+                    "YarpSensorBridge "
+                    "will not stream relevant measures.",
+                    logPrefix);
+    }
+
+    ret = m_pimpl->subConfigLoader("stream_current_sensors",
+                                   "Currents",
+                                   &YarpSensorBridge::Impl::configureRemoteControlBoardRemapper,
+                                   handler,
+                                   m_pimpl->metaData,
+                                   m_pimpl->metaData.bridgeOptions.isCurrentSensorsEnabled);
+    if (!ret)
+    {
+        log()->info("{} Skipping the configuration of configureRemoteControlBoardRemapper. "
+                    "YarpSensorBridge "
+                    "will not stream relevant measures.",
+                    logPrefix);
+    }
+
```

Comment:
I was wondering if there might be a cleaner way to do this, since within the `configureRemoteControlBoardRemapper` function, we clear the joint lists in the `metaData` everytime and read the same parameter from the configuration file.

Instead, we could just use part of the `subConfigLoader` logic somewhere enable non-default streams provided by the remote control board remapper? 

Open to discussion cc @GiulioRomualdi @S-Dafarra

Reply:
To be honest, I did not understand the problem

Reply:
The problem is quite minor.

[SubConfigLoader](https://github.com/dic-iit/bipedal-locomotion-framework/blob/71e37a822c9c7aa1bf23a91eeabceb6f66ba3165/src/RobotInterface/YarpImplementation/include/BipedalLocomotion/RobotInterface/YarpSensorBridgeImpl.h#L193-L228) is a helper function that checks for a `enable_stream` parameter (for example `enable_joint_states`) from the configuration file and then uses a `configureSomeTypeOfSensors` function pointer to load other associated parameters to this stream (in this case `configureRemoteControlBoardRemapper`).

Here, within the [`configureRemoteControlBoardRemapper`](https://github.com/dic-iit/bipedal-locomotion-framework/blob/71e37a822c9c7aa1bf23a91eeabceb6f66ba3165/src/RobotInterface/YarpImplementation/include/BipedalLocomotion/RobotInterface/YarpSensorBridgeImpl.h#L230-L266), the configuration that is being read is `jointsLists` only.  However, since now there other optional joints related quantities expected to be obtained from the same RemoteControlBoardRemapper, repeating this `subConfigLoader` calls basically tries to clear the `jointsList` and re-read the `jointsList` everytime. Maybe, it's ok to do so.. since we are only in the configuration step. But maybe, we could do better.

Reply:
To be honest, after reading the comments and the code several times, I had a hard time understanding the whole logic (clearly I am a bit slow :grin:). I wonder if it is possible to simplify the whole ``SubConfigLoader``/``configureRemoteControlBoardRemapper`` logic. If I understood well, the purpose of ``subConfigLoader`` was to reduce the use of repeated patterns. On the other hand, it is called repeatedly itself, while making the code much less readable. If it also has side effects, then it's becoming a little problematic.

Here is how I see it. 
Either we leave it as it is. Reading the same parameter a lot of times from the configuration parameters is not a drama. The complexity to read a configuration parameter is most probably linear in the ``yarp`` case. So it is not completely free, a bit of a waste, but not problematic.

Another possibility could be that instead of the ``SubConfigLoader``/``configureRemoteControlBoardRemapper`` logic we directly call a method that takes care of reading all the parameters, setting the relevant variables. 

What do you think?

Reply:
> To be honest, after reading the comments and the code several times, I had a hard time understanding the whole logic (clearly I am a bit slow grin). I wonder if it is possible to simplify the whole SubConfigLoader/configureRemoteControlBoardRemapper logic. If I understood well, the purpose of subConfigLoader was to reduce the use of repeated patterns. On the other hand, it is called repeatedly itself, while making the code much less readable. If it also has side effects, then it's becoming a little problematic.

To be honest, I took much more time than expected for this PR because of the same problem. I had a hard time understanding too (but consider that I'm super slow).

Reply:
> On the other hand, it is called repeatedly itself, while making the code much less readable. If it also has side effects, then it's becoming a little problematic.

I suppose the problem of readability arises only with the `RemoteControlBoardRemapper` since it exposes a variety of interfaces while being just one type of device. 

> Either we leave it as it is. Reading the same parameter a lot of times from the configuration parameters is not a drama. The complexity to read a configuration parameter is most probably linear in the yarp case. So it is not completely free, a bit of a waste, but not problematic.

This should be fine, I guess.

Reply:
Should we handle this in a different PR?
@prashanthr05 @GiulioRomualdi @S-Dafarra

Reply:
I'd say, since you're going to handle the suggested change in https://github.com/dic-iit/bipedal-locomotion-framework/pull/359#discussion_r667845184, it would be easier to handle this change as well.

But, It's anyways not blocking the prescribed functionality of this PR. So you can decide. :)

Reply:
I changed a bit the logic of `configureRemoteControlBoardRemapper`, `attachRemappedRemoteControlBoard` and `readControlBoardInterface` in `YarpSensorBridgeImpl.h`. I hope it is clear and it works fine. The commit is https://github.com/dic-iit/bipedal-locomotion-framework/pull/359/commits/a5f3bd21e818d1629b0ef4e482f63beb588559da.
@prashanthr05 @GiulioRomualdi @S-Dafarra

Reply:
The changes looks good. And also the check on the number of joints before reading the joint list parameter was a simple and effective solution.